arangod/Pregel/Worker/Handler.h

```diff
@@ -49,18 +50,22 @@ struct VerticesProcessed {
 
 template<typename V, typename E, typename M, typename Runtime>
 struct WorkerHandler : actor::HandlerBase<Runtime, WorkerState<V, E, M>> {
+  DispatchStatus const& dispatchStatus =
+      this->template dispatch<pregel::message::StatusMessages>;
+  DispatchMetrics const& dispatchMetrics =
+      this->template dispatch<metrics::message::MetricsMessages>;
+  DispatchConductor const& dispatchConductor =
+      this->template dispatch<conductor::message::ConductorMessages>;
```

Comment:
Just a thought: We could also only give the dispatchStatus to the receive function (instead of giving 3 dispatch function to it) because this is the only function that needs to be called during a state. The other dispatching - to metrics and conductor actor and also the calls to the status actor at the end of a state - could still be made inside the handler.
Pro: We just give one dispatch function to the state and therefore have less callbacks which can make the code more readable. The remaining sending is transparent in the handler. 
Con: Two different ways of dispatching from the worker - perhaps this is more confusing...
I'm tending a bit more to the approach with less callbacks, but are not sure what is the best approach, just wanted to mention it here.

Reply:
OK, let's see what @markuspf has to say.